### PR TITLE
refactor(upgrade): simplify return branches of GetUpgradeVersions

### PR DESCRIPTION
**Pull request by gunnar-solo**
# Description
Fixed a bug where `changelog` directory would always be dependent on `gloo` repository

_and_

offered a refactor of said method to make it easier to develop on

# Checklist:
- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: test/kube2e/upgrade/upgrade_utils.go [hunk 1]
```diff
@@ -15,72 +14,47 @@ import (
...
-	// so we can test this more easily.
-	client, err := githubutils.GetClient(ctx)
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, "unable to create github client")
+// This will return the (prevLtsRelease, latestRelease, err)
+// return possiblities (from best-case to worst-case):
+//   - 😃 (prevLtsRelease, latestRelease, nil)
+//   - 😐 (prevLtsRelease, nil, nil)
+//   - 😔 (nil, nil, err)
+func GetUpgradeVersions(ctx context.Context, repoName string) (*versionutils.Version, *versionutils.Version, error) {
```
Comment (gunnar-solo) [inline test/kube2e/upgrade/upgrade_utils.go:26]: since this is a full refactor of the `GetUpgradeVersions` method, I recommend not using the delta view

**Comment by gunnar-solo** [comment]
/kick https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=d983fa90-3d63-4d73-a37a-670ffefb6067

File: test/kube2e/upgrade/upgrade_utils.go [hunk 2]
```diff
@@ -15,72 +14,47 @@ import (
...
+	client, githubClientErr := githubutils.GetClient(ctx)
+	if githubClientErr != nil {
+		return nil, nil, errors.Wrapf(githubClientErr, "unable to create github client")
 	}
 
-	lastMinorLatestPatchVersion, lastMinorErr := getLatestReleasedPatchVersion(ctx, client, repoName, currentMinorLatestPatchVersion.Major, currentMinorLatestPatchVersion.Minor-1)
-	if lastMinorErr != nil {
-		// a true error lets return that.
-		return nil, nil, lastMinorErr
+	// get the latest and upcoming releases of the current branch
```
Comment (ianmacclancy) [inline test/kube2e/upgrade/upgrade_utils.go:39]: flow order nit: We get the files, get the git client, use the files, then use the github client. Let's get the files, use the files, get the github client, use the github client

File: test/kube2e/upgrade/upgrade_utils.go [hunk 3]
```diff
@@ -15,72 +14,47 @@ import (
...
-	}
-
-	// TODO(nfuden): Update goutils to not use a struct but rather interface
-	// so we can test this more easily.
-	client, err := githubutils.GetClient(ctx)
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, "unable to create github client")
+// This will return the (prevLtsRelease, latestRelease, err)
+// return possiblities (from best-case to worst-case):
+//   - 😃 (prevLtsRelease, latestRelease, nil)
```
Comment (ianmacclancy) [inline test/kube2e/upgrade/upgrade_utils.go:23]: Not sure what we are saying with these options. The mid option is actually an expected situation so would also be a smile?
If we want to break down the possible options we could be more informative 
- val, val, nil - we have all versions we want, no errors
- val, nil, nil - we are testing the first release of a minor version, no errors
- nil, nil, err - we are unable to fetch versions for upgrade tests

**Review by ianmacclancy — CHANGES_REQUESTED** [review]
Ordering and description changes

File: test/kube2e/upgrade/upgrade_utils.go [hunk 4]
```diff
@@ -15,72 +14,46 @@ import (
...
-		currentMinorLatestPatchVersion = currentBranch
-	} else if curMinorErr != nil {
-		return nil, nil, curMinorErr
+// GetUpgradeVersions returns two semantic versions of a repository:
+//   - prevLtsRelease: the latest patch release of v1.m-1.x
+//   - latestRelease:  the latest patch release of v1.m.x
+//
+// Originally intended for use in upgrade testing, it can return any of:
+//   - (prevLtsRelease, latestRelease, nil): all release versions computable
+//   - (prevLtsRelease, nil, nil):           only prevLtsRelease computable
```
Comment (bewebi) [inline test/kube2e/upgrade/upgrade_utils.go:26]:
```suggestion
//   - (prevLtsRelease, nil, nil):           only prevLtsRelease computable (ie current branch has never been released)
```
The inline comment explains this case, but I think it's helpful here as well
Feel free to reword

**Review by ianmacclancy — APPROVED** [review]

**Pull request merged**